l1-contracts/test/foundry/integration/GatewayTests.t.sol

```diff
@@ -219,6 +221,31 @@ contract GatewayTests is L1ContractDeployer, HyperchainDeployer, TokenDeployer,
         vm.stopBroadcast();
     }
 
+    function test_registerAlreadyDeployedHyperchain() public {
+        gatewayScript.registerGateway();
+        IStateTransitionManager stm = IStateTransitionManager(l1Script.getSTM());
+        IBridgehub bridgehub = IBridgehub(l1Script.getBridgehubProxyAddress());
+        address owner = Ownable(address(bridgeHub)).owner();
+
+        address chain = _deployZkChain(
+            currentHyperChainId++,
+            ETH_TOKEN_ADDRESS,
+            address(bridgehub.sharedBridge()),
+            owner,
+            stm.protocolVersion(),
+            stm.storedBatchZero()
+        );
+        address stmAddr = IZkSyncHyperchain(chain).getStateTransitionManager();
+
+        uint256 chainId = currentHyperChainId - 1;
+
+        vm.startBroadcast(owner);
+        bridgeHub.addStateTransitionManager(stmAddr);
+        bridgeHub.addTokenAssetId(DataEncoding.encodeNTVAssetId(chainId, ETH_TOKEN_ADDRESS));
+        bridgeHub.registerAlreadyDeployedHyperchain(chainId, chain);
+        vm.stopBroadcast();
```

Comment:
can you please double check that bridgehub's mappings were filled correctly? At least that the `getZKChain` / `getChainTypeManager`/`baseTokenAssetId` work